src/blk.rs

```diff
@@ -92,16 +92,26 @@ impl VirtIOBlk<'_> {
     /// **must not** spin on `resp` to wait for it to change. A safe way is to read it
     /// after the same token as this method returns is fetched through [VirtIOBlk::pop_used()],
     /// which means that the request has been ready.
-    pub fn read_block_nb(&mut self, block_id: usize, buf: &mut [u8], resp: &mut u8) -> Result<u16> {
+    ///
+    /// # Safety
+    ///
+    /// `buf` is still borrowed by the underlying virtio block device even if this
+    /// method returns. Thus, it is the caller's responsibility to guarantee that
+    /// `buf` is not accessed before the request is completed in order to avoid
+    /// data races.
+    pub unsafe fn read_block_nb(
+        &mut self,
+        block_id: usize,
+        buf: &mut [u8],
+        resp: &mut BlkResp,
```

Comment:
How does user initialize a `BlkResp` before calling it? Seems a `new` or `default` method is missing?

Reply:
Now I just use `BlkResp::default()`, which is also used in `read_block` and `write_block`.